Content.Server/Dragon/DragonSystem.cs

```diff
@@ -0,0 +1,195 @@
+using Content.Server.Body.Components;
+using Content.Server.Body.Systems;
+using Content.Server.Destructible;
+using Content.Server.DoAfter;
+using Content.Server.Popups;
+using Content.Shared.Actions;
+using Content.Shared.CharacterAppearance.Components;
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Damage;
+using Content.Shared.MobState;
+using Content.Shared.MobState.Components;
+using Content.Shared.Tag;
+using Robust.Shared.Audio;
+using Robust.Shared.Containers;
+using Robust.Shared.Player;
+using System.Threading;
+
+namespace Content.Server.Dragon
+{
+    public sealed class DragonSystem : EntitySystem
+    {
+        [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+        [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly BloodstreamSystem _bloodstreamSystem = default!;
+        [Dependency] private readonly SharedContainerSystem _containerSystem = default!;
+        
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<DragonComponent, ComponentStartup>(OnStartup);
+            SubscribeLocalEvent<DragonComponent, DevourActionEvent>(OnDevourAction);
+            SubscribeLocalEvent<DragonComponent, CarpBirthEvent>(OnCarpBirthAction);
+
+            SubscribeLocalEvent<TargetDevourSuccessfulEvent>(OnDevourSucsessful);
+            SubscribeLocalEvent<TargetDevourCancelledEvent>(OnDevourCancelled);
+            SubscribeLocalEvent<DragonComponent, MobStateChangedEvent>(OnMobStateChanged);
+        }
+
+        private void OnMobStateChanged(EntityUid uid, DragonComponent component, MobStateChangedEvent args)
+        {
+            //Empties the stomach upon death
+            //TODO: Do this when the dragon gets butchered instead
```

Comment:
is this a todo for a future PR or here? this shouldn't be too hard to do; butchering gibs the entity, and theres an event for before gibbing happens where you can handle this behavior.